crates/picky-krb/src/data_types.rs

```diff
@@ -152,28 +162,247 @@ pub type Ticket = ApplicationTag<TicketInner, 1>;
 ///         lr-value        [1] KerberosTime
 /// }
 /// ```
-#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
 pub struct LastReqInner {
     lr_type: ExplicitContextTag0<IntegerAsn1>,
     lr_value: ExplicitContextTag1<KerberosTime>,
 }
 pub type LastReq = Asn1SequenceOf<LastReqInner>;
 
+/// [MS-KILE 2.2.2](https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-KILE/%5bMS-KILE%5d.pdf)
+///
+/// ```not_rust
+/// KERB-ERROR-DATA ::= SEQUENCE {
+///     data-type [1] INTEGER,
+///     data-value [2] OCTET STRING OPTIONAL
+/// }
+/// ```
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct KerbErrorData {
+    data_type: ExplicitContextTag1<IntegerAsn1>,
+    #[serde(default)]
+    data_value: Optional<Option<ExplicitContextTag2<BitStringAsn1>>>,
```

Comment:
As I see you’ve been making fields public, I want to make sure that these are intentionally privates

Reply:
I made these fields also public